### PR TITLE
02-async-development

### PR DESCRIPTION
**Pull request by anzed**

### CODE REVIEW

File: modules/Importer.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* eslint arrow-body-style:0 */
+import csv from 'csvtojson';
+
+export default class Importer {
+    import(paths) {
```
Comment (boccob) [inline modules/Importer.js]: It should take only one path. It's a part of requirements :)
But you can leave it as is

File: modules/Importer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        const imports = paths.map((path) => {
+            return new Promise((resolve, reject) => {
+                csv().fromFile(path)
+                    .on('end_parsed', jsonObj => resolve(jsonObj))
+                    .on('done', error => reject(error));
+            });
+        });
+
+        return Promise.all(imports)
```
Comment (boccob) [inline modules/Importer.js]: But it does not return data from files. Also would be better to move console calls to `app.js`

File: modules/Importer.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            .then(data => console.log(data))
+            .catch(err => console.error(err));
+    }
+
+    importSync(paths) {
+        paths.forEach((path) => {
```
Comment (boccob) [inline modules/Importer.js]: It does not return anything.

File: modules/Dirwatcher.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+export default class Dirwatcher extends EventEmitter {
+    constructor(path, delay) {
+        super();
+
+        this.path = path;
+        this.delay = delay;
+    }
+
+    watch() {
```
Comment (boccob) [inline modules/Dirwatcher.js]: Did you read requirements? `watch` should take `path` and `delay`.

**Comment by boccob** [comment]
Well, it doesn't work for me at all. I see the log only when I run `npm start` but no info when I add /change / remove some data.

File: modules/Importer.js (hunk 2 above)
Comment (anzed) [reply modules/Importer.js]: Fixed

File: modules/Dirwatcher.js (hunk 4 above)
Comment (anzed) [reply modules/Dirwatcher.js]: Fixed

**Pull request merged**